sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -0,0 +1,1422 @@
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="inputs"></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response<DocumentResultCollection<LinkedEntity>>> ExtractEntityLinkingAsync(IEnumerable<DocumentInput> inputs, TextAnalyticsRequestOptions options, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(inputs, nameof(inputs));
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.AI.TextAnalytics.TextAnalyticsClient.ExtractEntityLinking");
+            scope.Start();
+
+            try
+            {
+                using Request request = CreateDocumentInputRequest(inputs, options, EntityLinkingRoute);
+                Response response = await _pipeline.SendRequestAsync(request, cancellationToken);
+
+                return response.Status switch
+                {
+                    200 => await CreateLinkedEntityResponseAsync(response, cancellationToken).ConfigureAwait(false),
+                    _ => throw await response.CreateRequestFailedExceptionAsync(),
+                };
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="inputs"></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual Response<DocumentResultCollection<LinkedEntity>> ExtractEntityLinking(IEnumerable<DocumentInput> inputs, TextAnalyticsRequestOptions options, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(inputs, nameof(inputs));
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.AI.TextAnalytics.TextAnalyticsClient.ExtractEntityLinking");
+            scope.Start();
+
+            try
+            {
+                using Request request = CreateDocumentInputRequest(inputs, options, EntityLinkingRoute);
+                Response response = _pipeline.SendRequest(request, cancellationToken);
+
+                return response.Status switch
+                {
+                    200 => CreateLinkedEntityResponse(response),
+                    _ => throw response.CreateRequestFailedException(),
+                };
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        #endregion
+
+        private Request CreateStringCollectionRequest(IEnumerable<string> inputs, string language, string route)
+        {
+            Argument.AssertNotNull(inputs, nameof(inputs));
```

Comment:
If already validated on public method input, this would be a waste of time. At best, could probably use Code Contracts for debug builds.